GrocifyApp.BLL/Implementations/ShoppingListService.cs

```diff
@@ -60,5 +60,21 @@ public async Task AddProductsToShoppingList(Guid id, Dictionary<Guid, ShoppingLi
 
             await _shoppingListProductRepository.SaveChangesAsync(token);
         }
+
+        public async Task<ShoppingList?> GetDefaultShoppingList()
+        {
+            return await repository.GetSingleWhere(b => b.DefaultList == true);
+        }
+
+        public async Task ChangeDefaultShoppingList(ShoppingList newDefault, ShoppingList actualDefault)
+        {
+            newDefault.DefaultList = true;
+
+            actualDefault.DefaultList = false;
```

Comment:
instead of getting actual defaultList and then updating to false, do this:

await repository.UpdateMultipleLeafType(x => x.HouseId == HouseId && x.DefaultList == true, x => x.SetProperty(y => y.DefaultList, false));

if you do this, it only performs 1 query to db instead of 2

Reply:
you can also do this:

var updatedEntities = await repository.UpdateMultipleLeafType(x => x.HouseId == HouseId && (x.DefaultList == true || x.Id == newDefault.Id),
    x => x.SetProperty(y => y.DefaultList, y => !y.DefaultList));


this will update both at the same time.
if updatedEntities = 2 => everything were fine.
else either no defaultList existed previously or newShoppingList doesn't exist

Reply:
in summary you have two options:

1)
getShoppingList to make default, throw if does not exist
if getShoppingList.DefaultList == true => return
update shoppingList that was previously default to false
update the new shoppingList to default

performed queries: 3

2)
get where DefaultList == true or Id = newShoppingListId
if(result.Count == 2 or (result.First().Id == newShoppingListId && result.First().DefaultList != true))
     update both old and new shopping list defaultList property at the same time
else
    throw NotFound

performed queries: 2